clients/rospy/src/rospy/impl/rosout.py

```diff
@@ -85,7 +85,20 @@ def _rosout(level, msg, fname, line, func):
                 try:
                     _in_rosout = True
                     msg = str(msg)
-                    topics = get_topic_manager().get_topics()
+
+                    # check parameter server/cache for omit_topics flag
+                    # the same parameter is checked in rosout_appender.cpp for the same purpose
+                    # parameter accesses are cached automatically in python
+                    if rospy.has_param("/rosout_disable_topics_generation"):
```

Comment:
This adds 1 or 2 XMLRPC calls to each log action (?)

Reply:
The comment indicates that the response is cached, but I agree that this is super bad if it's not the case.

Reply:
Looks like it got disabled:
https://github.com/ros/ros_comm/blob/e96c407c64e1c17b0dd2bb85b67f388380527097/clients/rospy/src/rospy/msproxy.py#L119

Reply:
Look a bit further:
https://github.com/ros/ros_comm/blob/e96c407c64e1c17b0dd2bb85b67f388380527097/clients/rospy/src/rospy/msproxy.py#L128